gift_card_custom/models/gift_card.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class GiftCardCustom(models.Model):
+    _inherit = "gift.card"
+
+    product_id = fields.Many2one('product.template', ondelete='cascade', string="Product", required=True, store=True, readonly=False)
```

Comment:
Le ondelete cascade me semble dangereux ici, ça vourdrais dire, quand on supprime l'article ça supprime la carte cadeau (bon si l'article est par ailleurs utilisé dans un bon de commande j'imagine que l'on ne pourra pas)

`store=True, readonly=False` sont les valeurs par défaut ici... mais j'imagine je suis pas encore allez lire le code des surcharge que j'imagine plus bas !

```suggestion
    product_id = fields.Many2one('product.template', string="Product", required=True,)
```